src/RouterFactory.php

```diff
@@ -27,18 +27,16 @@ public function __invoke(ContainerInterface $container): RouterInterface
         $factory = $this->engineFactory;
         /* @var $router RouterInterface */
         $router = $factory();
+        if (!$router->hasContainer()) {
```

Comment:
`hasContainer()` and `setContainer()` are not part of `RouterInterface`.

Reply:
Mmm... what is the decision?

Reply:
Could be solved by adding methods to `RouteCollectorInterface`.

Reply:
Done